### PR TITLE
Throw an error if structural simplification is applied twice

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
Fixes https://github.com/SciML/ModelingToolkit.jl/issues/3012

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
@YingboMa that failing test... doesn't make much sense?

**Comment by ChrisRackauckas** [comment]
Since downstreams all seem fine other than test bugs, this is good to go. I'm sure someone will complain about this one, but this is pretty much required in order to make sure no weird hidden bugs show up. Calling a system complete means you have completed the system: if you try to change it after you say you will never change it (i.e. that it's complete), bad things will happen. We have to assume users don't lie about when their system is completed or else we could never optimize things like schedules. So... this error is pretty necessary, and if you were violating it there's issues you could have hit before which is why it was made explicit.

**Comment by ChrisRackauckas** [comment]
I'll follow up with SBMLToolkit which is over zealous at calling `complete` in the system construction which is simply invalid: a file reader cannot read the future on what a user wants to do.

**Comment by baggepinnen** [comment]
>  I'm sure someone will complain about this one

yes I will, this was quite breaking. Calling `complete` is required to access variables with the correct name space, and one has to access variables before `structural_simplify` for things like linearization and generation of dynamics functions with inputs. Why is `iscomplete` used for this check and not a check that actually determines if `structural_simplify` has been called?

**Comment by ChrisRackauckas** [comment]
`iscomplete` is a flag for whether no more changes will happen to a system. You're literally saying that you are breaking that promise, and there are many things that rely on that promise. Of course you want an error when that happens!

File: src/systems/systems.jl [hunk 1]
```diff
@@ -21,6 +29,7 @@ function structural_simplify(
         sys::AbstractSystem, io = nothing; simplify = false, split = true,
         allow_symbolic = false, allow_parameter = true, conservative = false, fully_determined = true,
         kwargs...)
+    iscomplete(sys) && throw(RepeatedStructuralSimplificationError())
```
Comment (YingboMa) [inline src/systems/systems.jl:32]: This is the wrong check and breaking. You should just check the existence of schedule.

**Pull request merged**